packages/backend/src/services/SavedSqlService/SavedSqlService.ts

```diff
@@ -29,12 +31,53 @@ export class SavedSqlService extends BaseService {
 
     private readonly savedSqlModel: SavedSqlModel;
 
+    private readonly schedulerClient: SchedulerClient;
+
     constructor(args: SavedSqlServiceArguments) {
         super();
         this.analytics = args.analytics;
         this.projectModel = args.projectModel;
         this.spaceModel = args.spaceModel;
         this.savedSqlModel = args.savedSqlModel;
+        this.schedulerClient = args.schedulerClient;
+    }
+
+    private async hasAccess(
```

Comment:
Move duplicated code to private methods + Update all methods to use `hasSavedChartAccess` or `hasAccess`.